### PR TITLE
powered by wazimap

### PR DESCRIPTION
**Pull request by emre2038**
## Description
* narrow enough to fit inside rich data in page nav
* hidden when rich data in page nav is minimised so that it doesn't overlap with rich data content
* data mapper and point mapper menus have space at the bottom so that it doesn't overlap their content, their scroll space ends above it

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-555

## How is it tested automatically?
added `__tests__/gui/watermark.feature`

## How should a reviewer test it locally
* confirm that the watermarks are always in the same position
* confirm that in the data mapper and the point mapper, the scroll area ends above the watermark. and the watermark is always visible
* confirm that in the rich data panel, if you collapse the categories menu the watermark is not visible anymore

## Screenshots
* rich data
image

* rich data with the categories menu collapsed
image

* data mapper
image

* data mapper with the scrollbar
image

* point mapper
image

* point mapper with the scrollbar
image

### Added
* `src/js/ui_components/watermark.js` watermark react component
* `src/custom-css/general.scss` for site-wide css changes

### Updated
* `src/js/profile/profile_loader.js` to add the watermark to rich data
* `src/js/map/maps.js` to add the watermark to the map
* `src/js/elements/point_tray/tray.js` to add the watermark to the point mapper
* `src/js/elements/data_mapper/menu.js` to add the watermark to the data mapper
* `src/js/webflow/import.js` to set up the watermark elements in the index.html

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally
- [x] 👩‍🎨 does the design match the [Demo](https://wazimap-ng-v1.webflow.io/demo)

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out
- [x] commit messages are meaningful

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers

### Testing

- [x]  ✅ added (appropriate) unit tests
- [x]  💢 edge cases in tests were considered
- [x]  ✅ ran tests locally & are passing

### CODE REVIEW

File: __tests__/gui/watermark.feature [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    And I click on "Test Theme" in Point Mapper
+    And I click on "Higher Education" in Point Mapper
+    And I click on "Labour" in Point Mapper
+    Then I confirm that the point mapper watermark is visible
+
+    # Rich data
+    And I expand Rich Data
+    Then I confirm that the rich data watermark is visible
+    And I collapse Rich Data Nav
+    Then I confirm that the rich data watermark is not visible
```
Comment (michaelglenister) [inline __tests__/gui/watermark.feature:37]: I think we want one more line to test that watermark is still visible after hiding rich data watermark, such as
`Then I confirm that the map watermark is visible`

File: src/js/map/maps.js [hunk 2]
```diff
@@ -40,10 +43,23 @@ export class MapControl extends Component {
...
 
+    addWatermark() {
+        let watermarkWrapper = document.createElement('div');
+        $(watermarkWrapper).addClass('watermark-wrapper');
+        $('.map-watermark-wrapper')
+            .append(watermarkWrapper);
+
+        let watermarkRoot = createRoot(watermarkWrapper);
+        watermarkRoot.render(<Watermark/>);
+    }
```
Comment (michaelglenister) [inline src/js/map/maps.js:61]: This block of code is repeated in tray,js and menu.js, could we add a single function to call from each place?

File: src/js/map/maps.js [hunk 3]
```diff
@@ -64,8 +80,6 @@ export class MapControl extends Component {
             .map('main-map', mapOptions.leafletOptions)
             .setView([coords["lat"], coords["long"]], coords["zoom"])
 
-        map.attributionControl.addAttribution('&copy; <a href="https://www.openstreetmap.org/#map=6/-28.676/24.677" target="_blank">OpenStreetMap</a> contributors, <a href="https://carto.com/help/working-with-data/attribution/#basemaps" target="_blank">Carto</a>. Powered by <a href="https://openup.org.za/products/wazimap-ng" target="_blank">Wazimap</a>');
```
Comment (michaelglenister) [inline src/js/map/maps.js:67]: This removed attribution for Open Street Map and Carto, does their Terms of Use allow us not to attribute them?

If this is okay the boundary selection and zoom controls at the bottom right need to be aligned in some way

File: src/js/profile/profile_loader.js [hunk 4]
```diff
@@ -87,8 +95,37 @@ export default class ProfileLoader extends Component {
...
+
+    addWatermark() {
+        if ($('.rich-data-nav .watermark-wrapper').length > 0) {
+            return;
+        }
+
+        let watermarkWrapper = document.createElement('div');
+        $(watermarkWrapper)
+            .addClass('watermark-wrapper')
+            .addClass('truncate');
```
Comment (michaelglenister) [inline src/js/profile/profile_loader.js:112]: `.addClass('truncate');`

Oh, that small change is going to make it challenging to separate into a single class. Perhaps it's okay to leave the add watermark logic separate for each element

**Review by michaelglenister — COMMENTED** [review]
Looks good, just a few small changes

File: __tests__/gui/watermark.feature (hunk 1 above)
Comment (emre2038) [reply __tests__/gui/watermark.feature]: done 👍

**Review by michaelglenister — APPROVED** [review]

**Pull request merged**